common/beerocks/bwl/dummy/base_wlan_hal_dummy.cpp

```diff
@@ -280,9 +280,23 @@ bool base_wlan_hal_dummy::set(const std::string &param, const std::string &value
 
 bool base_wlan_hal_dummy::ping() { return true; }
 
-bool base_wlan_hal_dummy::dummy_send_cmd(const std::string &cmd) { return false; }
-
-bool base_wlan_hal_dummy::dummy_send_cmd(const std::string &cmd, char **reply) { return false; }
+bool base_wlan_hal_dummy::write_status_file(const std::string &filename, const std::string &value)
+{
+    auto full_path{get_status_dir() + "/" + filename};
+    auto full_path_tmp{full_path + ".tmp"};
```

Comment:
Best practice (IIRC) is to use `mkstemp` (or similar) for temporary files. No issues with concurrency, security (e.g. people guessing and pre-opening/redirecting the file), file is guaranteed to be on a writable FS, etc

Reply:
mkstemp has an annoying interface...

Concurrency is not an issue here, if multiple agents are running, we have bigger problems.

Security is also not really an issue since it's only used for testing.

Guaranteed on a writable FS: actually no, because we have to create it in the same FS as where it goes in the end, otherwise the rename is not atomic.

So I choose to ignore this comment. If you insist, I can still do it. Actually maybe it's not that much work, but IMO not really a priority now.